service/src/main/java/com/ge/predix/acs/GlobalControllerAdvice.java

```diff
@@ -0,0 +1,28 @@
+package com.ge.predix.acs;
+
+import com.ge.predix.acs.privilege.management.PrivilegeManagementUtility;
+import com.ge.predix.acs.privilege.management.ZoneDoesNotExistException;
+import org.json.simple.JSONObject;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+
+@ControllerAdvice
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class GlobalControllerAdvice {
+    @ExceptionHandler(ZoneDoesNotExistException.class)
+    public ResponseEntity<JSONObject> handleZoneDoesNotExist() {
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).contentType(MediaType.APPLICATION_JSON)
```

Comment:
After more thought this should probably be a ```BAD REQUEST```. This is because when we are using ACS with uaa-token-lib for the predix profile, token lib returns a 400 if the zone doesn't exist in ZAC. To be consistent with that behavior, we should probably switch this.